packages/twenty-front/src/modules/object-record/record-field/form-types/hooks/useTextVariableEditor.ts

```diff
@@ -0,0 +1,78 @@
+import { initializeEditorContent } from '@/workflow/search-variables/utils/initializeEditorContent';
+import { VariableTag } from '@/workflow/search-variables/utils/variableTag';
+import Document from '@tiptap/extension-document';
+import HardBreak from '@tiptap/extension-hard-break';
+import Paragraph from '@tiptap/extension-paragraph';
+import { default as Placeholder } from '@tiptap/extension-placeholder';
+import Text from '@tiptap/extension-text';
+import { Editor, useEditor } from '@tiptap/react';
+import { isDefined } from 'twenty-ui';
+
+type UseTextVariableEditorProps = {
+  placeholder: string | undefined;
+  multiline: boolean | undefined;
+  readonly: boolean | undefined;
+  defaultValue: string | undefined;
+  onUpdate: (editor: Editor) => void;
+};
+
+export const useTextVariableEditor = ({
+  placeholder,
+  multiline,
+  readonly,
+  defaultValue,
+  onUpdate,
+}: UseTextVariableEditorProps) => {
+  const editor = useEditor({
+    extensions: [
+      Document,
+      Paragraph,
+      Text,
+      Placeholder.configure({
+        placeholder,
+      }),
+      VariableTag,
+      ...(multiline
+        ? [
+            HardBreak.configure({
+              keepMarks: false,
+            }),
+          ]
+        : []),
+    ],
+    editable: !readonly,
+    onCreate: ({ editor }) => {
+      if (isDefined(defaultValue)) {
+        initializeEditorContent(editor, defaultValue);
+      }
+    },
+    onUpdate: ({ editor }) => {
+      onUpdate(editor);
+    },
+    editorProps: {
+      handleKeyDown: (view, event) => {
+        if (event.key === 'Enter' && !event.shiftKey) {
```

Comment:
Could this conflict with our useScopedHotkeys ? It should be more appropriate to add a hotkey scope for this and call useScopedHotkeys here no ?

Reply:
I'm not sure how we can integrate hotkey scopes with tiptap here. I agree that it might be better to use hotkey scopes. We should see who we can do that. Maybe @thomtrp has already tried to implement that?

Reply:
If tip tap doesn't allow us to take precedence over its listeners, maybe we could at least set a hotkey scope and check with a simple if whether the hotkey scope is activated for this field ?

Reply:
That way we avoid conflicts, especially on forms where there are multiple inputs listening at the same time

Reply:
I created a separate issue to track that refactor: https://github.com/twentyhq/twenty/issues/8786.